### DISCUSSION TITLE
Was urine output really measured hourly? [MIMIC-IV]

### DISCUSSION THREAD

**Discussion by Qinyu-Allen-Zhao** [MIT-LCP/mimic-code#1140] [MIMIC-IV]
Hi!

I learned the KDIGO_uo.sql and also carefully read #476, but I think there is evidence that urine output maybe isn't recorded hourly. 

The kdigo-uo query is based on the assumption that data measured corresponds to urine output during the last hour. But I found some documentation was not hourly. I counted the number of those "special" records:
```
SELECT 
    SUM(CASE WHEN charttime != DATE_TRUNC(charttime, HOUR) THEN 1 ELSE 0 END) AS special
    , count(*) AS total_num
    , ROUND(SUM(CASE WHEN charttime = DATE_TRUNC(charttime, HOUR) THEN 1 ELSE 0 END) / count(*) *100, 2) AS ratio
FROM `physionet-data.mimic_derived.urine_output`
```

The result is:
Line |  special | total_num | ratio
-- | -- | -- | --
1 | 375347 | 3497267 | 10.73

That means about 10% of documentation is not hourly, for example, at 3:20 instead of 3:00.

Besides, I found that much missing documentation. Take stay_id 30367285 as an example
```
SELECT 
    *
FROM `physionet-data.mimic_derived.urine_output`
WHERE stay_id = 30367285
ORDER BY charttime
```

![image](https://user-images.githubusercontent.com/71951741/129570736-73b12360-dc18-4d2c-be5e-fbbeda560cec.png)

There are no records from 18:00 to 20:00, but there is a row of data at 20:31.
If the documentation is hourly, the data between 18 and 20 were missing?

Do those mean in fact, the previous charttime is a start time for the current UO measure? If so, these results can be explained I suppose.

Many thanks!

**Comment by SichengH** [comment]
Here is the code for generating that derived data. 
https://github.com/MIT-LCP/mimic-code/blob/main/mimic-iv/concepts/measurement/urine_output.sql

Seems the "charttime" that not showing as whole hours are inherited from the raw data. Also not every hour has urine output data. I didn't find any systamtic missings when convert raw data to derived data.

**Reply by Qinyu-Allen-Zhao** [reply]
Now that there are no missings, does that mean "each UO measure is the output volume from the last to the current charttime"?
For example, if a patient has these records:
   5:00 80
   6:00 90
   8:00 100
So the 100 at 8 o'clock is the UO from 6 to 8?

But according to the code for generating KDIGO stages (in both MIMIC-III and -IV), "[you] assume data charted at charttime corresponds to 1 hour of UO".
https://github.com/MIT-LCP/mimic-code/blob/main/mimic-iv/concepts/organfailure/kdigo_uo.sql
In this way, 100 at 8 is the UO from 7 to 8, but the UO from 6 to 7 is missing, isn't it?

I'm confused about which explanation is the case. I have only access to the data but have no idea about how the data were recorded. Please help me with that! Many thanks!
